src/http/oauth1.ts

```diff
@@ -99,24 +102,60 @@ export function createOAuth1HttpClient(
     tokenData = data;
   }
 
-  httpWithAuth.interceptors.request.use(async (config = {}) => ({
-    ...config,
-    headers: {
-      ...config.headers,
-      'Content-Type': 'application/json',
-      ...(config?.method
-        ? oAuth.toHeader(
-            oAuth.authorize(
-              {
-                url: `${config.baseURL}${config.url}`,
-                method: config.method,
-              },
-              tokenData
-            )
-          )
-        : {}),
-    },
-  }));
+  httpWithAuth.interceptors.request.use(async (config = {}) => {
+    const decodedUrl = `${config.baseURL}${config.url}`;
+    const { protocol, host, pathname, search } = new URL(decodedUrl);
+
+    const baseUrl = `${protocol}//${host}${pathname}`;
+    const searchParameters = qs.parse(
+      search.startsWith('?') ? search.slice(1) : search
+    );
+
+    const nonce = generateNonce();
+    const timeStamp = getTimeStamp();
+
+    const parameters = {
+      oauth_signature_method: 'HMAC-SHA1',
+      oauth_timestamp: timeStamp,
+      oauth_nonce: nonce,
+      oauth_consumer_key: oAuth.consumer.key,
+      oauth_token: tokenData.token,
+      oauth_version: '1.0',
+    };
+
+    const signatureParameters = {
+      ...searchParameters,
+      ...omit(parameters, ['realm', 'oauth_realm', 'oauth_signature']),
+    };
+
+    const signature = oauthSign.sign(
```

Comment:
All of the code is written to match the input of these boys
https://github.com/ExtraHorizon/auth-service/blob/dev/src/lib/oauth1Signature.ts